src/resolvelib/resolvers.py

```diff
@@ -383,6 +414,17 @@ def resolve(self, requirements, max_rounds):
                 if not success:
                     raise ResolutionImpossible(self.state.backtrack_causes)
             else:
+                # discard as information sources any invalidated names
+                # (unsatisfied names that were previously satisfied)
+                newly_unsatisfied_names = {
+                    key
+                    for key, criterion in self.state.criteria.items()
+                    if key in satisfied_names
+                    if not self._is_current_pin_satisfying(key, criterion)
```

Comment:
```suggestion
                    and not self._is_current_pin_satisfying(key, criterion)
```
Maybe you mean this?

Reply:
They're equivalent actually.